crates/rome_js_analyze/src/analyzers/nursery/no_useless_switch_case.rs

```diff
@@ -0,0 +1,155 @@
+use rome_analyze::context::RuleContext;
+use rome_analyze::{declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic};
+use rome_console::markup;
+use rome_diagnostics::Applicability;
+use rome_js_syntax::{JsCaseClause, JsDefaultClause};
+use rome_rowan::{AstNode, AstNodeList, BatchMutationExt, Direction, SyntaxElement};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow useless case in switch statements.
+    ///
+    /// A switch statement can optionally have a default clause.
+    ///
+    /// The default clause will be still executed only if there is no match in the case clauses.
+    /// An empty case clause that precedes the default clause is thus useless.
+    ///
+    /// Source: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-useless-switch-case.md
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (foo) {
+    ///     case 0:
+    ///     default:
+    ///         break;
+    ///     case 1:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// switch (foo) {
+    ///     default:
+    ///     case 0:
+    ///         break;
+    ///     case 1:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// switch (foo) {
+    ///     case 0:
+    ///         break;
+    ///     default:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// switch (foo) {
+    ///     case 0:
+    ///         break;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUselessSwitchCase {
+        version: "11.0.0",
+        name: "noUselessSwitchCase",
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessSwitchCase {
+    type Query = Ast<JsDefaultClause>;
+    type State = JsCaseClause;
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let default_clause = ctx.query();
+        let it = default_clause
+            .syntax()
+            .siblings(Direction::Prev)
+            .filter_map(JsCaseClause::cast)
+            .take_while(|case| case.consequent().len() == 0);
+        if default_clause.consequent().len() == 0 {
+            it.chain(
+                default_clause
+                    .syntax()
+                    .siblings(Direction::Next)
+                    .filter_map(JsCaseClause::cast)
+                    .take_while(|case| case.consequent().len() == 0),
+            )
+            .chain(
+                default_clause
+                    .syntax()
+                    .siblings(Direction::Next)
+                    .filter_map(JsCaseClause::cast)
+                    .find(|case| case.consequent().len() > 0),
+            )
+            .collect::<Vec<_>>()
```

Comment:
Nit: The turbo fish annotation shouldn't be necessary here because Rust can infer the type from the return type annotation.

```suggestion
.collect()
```